mernarezk/assignment2/Q2.java

```diff
@@ -0,0 +1,28 @@
+package Assignment2;
+
+public class Q2 {
+	/**
+	 * Returns reference to lowest common ancestor of p and q vals. This method
+	 * assumes that there are no duplicates in the tree also assumes that both p and
+	 * q are always present in the tree. if either p or q matches with current root
+	 * val, then we return that this is our current LCA other wise we try to search
+	 * for it in left subtree and right subtree if any of them return non-null
+	 * answer and other returns null then current root is LCA.
+	 * 
+	 * @param root
+	 * @param p
+	 *            first value that we need its common ancestor.
+	 * @param q
+	 *            second value that we need its common ancestor.
+	 * @return
+	 */
+	public static TreeNode commonAncestor(TreeNode root, int p, int q) {

```

Comment:
Very nice! This solution is different to the one we've discussed on Friday, but it is also O(N) from what I can tell, and it also looks very elegant. Good job! :)

Reply:
Thank you! I will check the solutions for the Friday solution too and let you know if it's not clear for me

Reply:
Oh I really like this solution too! :)